### PR TITLE
[9.0] base_view_inheritance_extension: Implement target_position for position="move"

### PR DESCRIPTION
**Pull request by mehdi-ghezal**
By default, inheritance with spec **position="move"** append the node inside the target. 
The spec **target_position** let you choose what you want to do : 
- **inside**: append the node inside the target
- **before**: append the node before target
- **after**: append the node after target

By default, if no value is provide we use the standard behaviour **inside**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Thanks for this feature!

Please change module version number according https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers

**Review by hbrunn — CHANGES_REQUESTED** [review]
runbot has a few complaints, and please remove the line about this from the roadmap then too. Also mention in the readme that this works so that other devs don't have to study the code.

you should also extend the tests so that all your code branches are hit by testing

**Review by hbrunn — CHANGES_REQUESTED** [review]
this is already great. Now just add your case to https://github.com/mehdi-ghezal/server-tools/blob/d5bcd02230124fd75bdfce1608d1930be9fd8018/base_view_inheritance_extension/demo/ir_ui_view.xml and assert something sensible about it in https://github.com/mehdi-ghezal/server-tools/blob/d5bcd02230124fd75bdfce1608d1930be9fd8018/base_view_inheritance_extension/tests/test_base_view_inheritance_extension.py

**Comment by StefanRijnhart** [comment]
Hi @mehdi-ghezal, can you please send your CLA to the OCA and add the extra tests? That way we can adopt this useful feature. Let us know if you need help.

FYI I'm now using this 10.0 version https://github.com/StefanRijnhart/server-tools/tree/feature/10.0/base_view_inheritance_extension/target_position_mehdi_ghezal

File: base_view_inheritance_extension/README.rst
Comment (Garamotte) [inline base_view_inheritance_extension/README.rst:44]: If this means `simply create a new node around an existing node`, this is already possible with the standard `replace`, [using `$0`](https://www.odoo.com/documentation/11.0/reference/views.html#inheritance-specs).

If this means `create a new node anywhere and move another node in it in a single xpath node`, I think the same logic could be used on the `move` position to do it with a single xpath node.

Comment (hbrunn) [reply base_view_inheritance_extension/README.rst:44]: we're in 9, `$0` is a feature from 10 upwards. Counterintuitive as this is, it might indeed make sense to backport this feature instead of cooking up another syntax

Comment (Garamotte) [reply base_view_inheritance_extension/README.rst:44]: Oh, you're right, I didn't took attention to the target version :)

File: base_view_inheritance_extension/models/ir_ui_view.py
Comment (NL66278) [inline base_view_inheritance_extension/models/ir_ui_view.py:130]: Please support 'move' as a (maybe deprecated) synonym for 'inside'. This module is already used a lot, with the move value. So better not break existing views.

Comment (StefanRijnhart) [reply base_view_inheritance_extension/models/ir_ui_view.py:130]: @NL66278 This is an additional attribute next to `position='move'` which remains necessary.

**Comment by hbrunn** [comment]
@StefanRijnhart are you waiting for this to be merged to propose your port to 10?

**Review by hbrunn — APPROVED** [review]

**Comment by StefanRijnhart** [comment]
@hbrunn I'm waiting for the CLA clearance before I propose my port.

**Comment by hbrunn** [comment]
given the time this takes already it might not happen at all, maybe @NL66278 could reimplement this in a clean room fashion and propose himself?

**Comment by StefanRijnhart** [comment]
That would be a solution. I share the fear that we are in a dead corner.

**Comment by mehdi-ghezal** [comment]
Hi, I'm sorry but I am quite busy, I just sent the CLA by email :)

**Comment by hbrunn** [comment]
@great. then the cla bot will kick in in a while and that's fine. But we still need to increase code coverage, please check https://github.com/OCA/server-tools/pull/1056#pullrequestreview-80411060 where I propose how to add tests

File: base_view_inheritance_extension/models/ir_ui_view.py
Comment (NL66278) [reply base_view_inheritance_extension/models/ir_ui_view.py:130]: Yes I see, I was put on a wrong footing by the README, which does not mention target_position.

**Comment by hbrunn** [comment]
yes, please also make it clear in the readme that it's about attribute `target_position`

**Comment by remihb** [comment]
LGTM, PR still pending due to missing tests ?

I really need this feature and I don't want to use my own fork anymore as the PR is working as expected since a while (tested in production in 9.0 and 10.0)

**Comment by hbrunn** [comment]
missing tests and not enough approvals

**Comment by remihb** [comment]
Having my approval is not enought ? :)
Ok, I will set up some tests and hope..

**Comment by hbrunn** [comment]
you didn't use the review/approve mechanism, so you don't count yet

**Review by NL66278 — APPROVED** [review]

**Review by remihb — APPROVED** [review]

**Comment by remihb** [comment]
Ok, I wasn't sure of the process. I thought that because of missing tests, the PR shouldn't be approved.

**Comment by hbrunn** [comment]
it's fine to approve, the label `ready to merge` will only appear when all tests succeed, in this case, when coverage is sufficiently high

**Comment by StefanRijnhart** [comment]
Did the cla-bot come back with a green light? I don't see it.

**Comment by remihb** [comment]
Ok, what can I do if I want to add tests above mehdi's changes? Should I make my own PR based on this one?

**Comment by StefanRijnhart** [comment]
OK, I checked the OCA database and there is a CLA! Yes @remihb, in that case please create PR to replace this one.

**Pull request closed without merging**
